server/game/core/ability/CardAbility.js

```diff
@@ -140,7 +135,7 @@ class CardAbility extends CardAbilityStep {
     displayMessage(context, messageVerb = context.source.isEvent() ? 'plays' : 'uses') {
         if (
             context.source.isEvent() &&
-            context.source.location !== Location.Hand
+            context.source.location !== Location.Discard
```

Comment:
What's the cause and/or impact of this change?

Reply:
It has to do with the fact that we now bypass abilities that have no effect on game state to reduce UX noise. What happened is that events which are no-ops were getting bypassed, but the effect that moved the event card to discard was actually happening as part of the event ability resolution (not the play event card resolution). So it was getting skipped if we skipped the ability resolution.

So I moved the event card -> discard move a little earlier in the pipeline and now it works. It's just that some checks were expecting event cards to be in hand or in a special play zone since that's how it worked in l5r.